### PR TITLE
Add technical advisory 133479

### PR DESCRIPTION
**Pull request by mikeCRL**

### CODE REVIEW

File: src/current/advisories/a133479.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+If these values were included in diagnostic reporting sent to Cockroach Labs, they were stored in a secure, non-public, and restricted Cockroach Labs telemetry database and have now been deleted. If you need help determining if this information was sent to Cockroach Labs by your CockroachDB cluster(s), email security@cockroachlabs.com with the subject "TA 133479 - Cluster Review" and a list of the Cluster IDs in question, and we will respond after evaluating what data we may have received.
+
+## Statement
+
+The default cluster setting redaction behavior in diagnostics will be resolved in CockroachDB versions 23.1.29, 23.2.14, 24.1.7, and 24.2.5.
+
+## Mitigation
+
+Users of CockroachDB versions 20.2, 21.1, 21.2, 22.1, 22.2, 23.1.0 - 23.1.28, and 23.2.0 - 23.2.13 are encouraged to:
```
Comment (mdlinville) [inline src/current/advisories/a133479.md:38]: This is missing the v24.2 impacted versions. Can we sort the affected versions from newest to oldest? We could even list the EOL versions separately from the supported versions, to provide yet another prompt to upgrade.

**Review by mdlinville — APPROVED** [review]

File: src/current/advisories/a133479.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+- `server.oidc_authentication.client_id`
+
+The affected cluster settings in versions 23.2.1+, 24.1, and 24.2 are:
+
+- `server.oidc_authentication.client_id`
+- `server.oidc_authentication.client_secret`
+
+To prevent the values of these cluster settings from being sent, we recommend either upgrading to specific patch versions as they become available (as detailed in the [Statement](#statement) section) or following the [Mitigation](#mitigation) steps outlined below. However, even if you do not correct this issue and continue to send this data to Cockroach Labs, it will be automatically discarded by our systems, if received.
+
+If these values were included in diagnostic reporting sent to Cockroach Labs, they were stored in a secure, non-public, and restricted Cockroach Labs telemetry database and have now been deleted. If you need help determining if this information was sent to Cockroach Labs by your CockroachDB cluster(s), email security@cockroachlabs.com with the subject "TA 133479 - Cluster Review" and a list of the Cluster IDs in question, and we will respond after evaluating what data we may have received.
```
[the remark this answers is not in the document]
Comment (mdlinville) [reply src/current/advisories/a133479.md:30]: I didn't test the mailto in a local build, so maybe confirm that it really does set the subject correctly.

File: src/current/advisories/a133479.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+affected_versions: v20.2, v21.1, v21.2, v22.1, v22.2, v23.1.0 to v23.1.28, v23.2.0 to 23.2.13, v24.1.0 to v24.1.6, v24.2.0 to 24.2.4
+advisory_date: 2024-10-25
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+CockroachDB diagnostics reporting, if enabled for a cluster in CockroachDB Cloud or in a Self-Hosted deployment, could have been sending the values of the cluster settings specified below through the telemetry payload sent to Cockroach Labs or stored by you locally. This behavior was due to a bug that incorrectly used user-configurable cluster setting redaction code paths when reporting diagnostics. This data is no longer stored by Cockroach Labs and is no longer sent in upcoming patch releases.
```
[the remark this answers is not in the document]
Comment (mdlinville) [reply src/current/advisories/a133479.md:15]: Without an edit, it is a little hard to parse "no longer sent in upcoming patch releases".

File: src/current/advisories/a133479.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
+title: Technical Advisory 133479
+advisory: A-133479
+summary: Diagnostics reporting, if enabled for a cluster, may have sent OIDC credentials stored in cluster settings to a secure, limited access Cockroach Labs telemetry database. The payload would have contained a unique cluster ID, and no publicly identifiable customer information.
```
[the remark this answers is not in the document]
Comment (mikeCRL) [reply src/current/advisories/a133479.md:4]: Leaving this one as is

File: src/current/advisories/a133479.md [hunk 5]
```diff
@@ -12,36 +12,36 @@ Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
 
 ## Description
 
-CockroachDB diagnostics reporting, if enabled for a cluster in CockroachDB Cloud or in a Self-Hosted deployment, could have been sending the values of the cluster settings specified below through the telemetry payload sent to Cockroach Labs or stored by you locally. This behavior was due to a bug that incorrectly used user-configurable cluster setting redaction code paths when reporting diagnostics. This data is no longer stored by Cockroach Labs and is no longer sent in upcoming patch releases.
+CockroachDB diagnostics reporting, if enabled for a cluster in CockroachDB Cloud or in a Self-Hosted deployment, could have been sending the values of the cluster settings specified below through the telemetry payload sent to Cockroach Labs or stored by you locally. This behavior was due to a bug that incorrectly used user-configurable cluster setting redaction code paths when reporting diagnostics. This data is no longer stored by Cockroach Labs and is no longer sent after the cluster is upgraded to a patch release with the fix.
```
Comment (mikeCRL) [inline src/current/advisories/a133479.md:15]:
```suggestion
CockroachDB diagnostics reporting, if enabled for a cluster in CockroachDB Cloud or in a Self-Hosted deployment, could have been sending the values of the cluster settings specified below through the telemetry payload sent to Cockroach Labs or stored by you locally. This behavior was due to a bug that incorrectly used user-configurable cluster setting redaction code paths when reporting diagnostics. This data is no longer stored by Cockroach Labs and is no longer sent after the cluster is upgraded to a patch release with the fix or after manual intervention via a cluster setting change.
```

File: src/current/advisories/a133479.md [hunk 6]
```diff
@@ -14,34 +14,34 @@ Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
...
 
-If these values were included in diagnostic reporting sent to Cockroach Labs, they were stored in a secure, non-public, and restricted Cockroach Labs telemetry database and have now been deleted. If you need help determining if this information was sent to Cockroach Labs by your CockroachDB cluster(s), email security@cockroachlabs.com with the subject "TA 133479 - Cluster Review" and a list of the Cluster IDs in question, and we will respond after evaluating what data we may have received.
+If these values were included in diagnostic reporting sent to Cockroach Labs, they were stored in a secure, non-public, restricted Cockroach Labs telemetry database, and have now been deleted. If you need help determining if this information was sent to Cockroach Labs by your CockroachDB clusters, email <a href="mailto:security@cockroachlabs.com?subject=TA 133479 Cluster Review">security@cockroachlabs.com</a> with the subject "TA 133479 - Cluster Review" with a list of the Cluster IDs in question, and we will respond after evaluating what data we may have received.
```
Comment (mikeCRL) [inline src/current/advisories/a133479.md:30]:
```suggestion
If these values were included in diagnostic reporting sent to Cockroach Labs, they were stored in a secure, non-public, restricted Cockroach Labs telemetry database, and have now been deleted. If you need help determining if this information was sent to Cockroach Labs by your CockroachDB clusters, email <a href="mailto:security@cockroachlabs.com?subject=TA 133479 Cluster Review">security@cockroachlabs.com</a> with the subject "TA 133479 - Cluster Review" and a list of the Cluster IDs in question, and we will respond after evaluating what data we may have received.
```

**Review by mdlinville — APPROVED** [review]
